### ISSUE TITLE
GDPR consent not working on Android

### ISSUE THREAD

**Issue by SharyRajpoot** [mopub/mopub-android-sdk#450]
I am using Mopub SDK 5.15.0 currently the latest one. BUT
On the device, I installed a VPN and choose different EU countries, but every time I launch the app always get the same. isgdpr applicable => false. I tried then AdMob consents they are working fine, but Mopub is not working in this case.

Does anyone have the same problem or any solution except forcedGdrp?

**Comment by hmertkahraman** [comment]
Hi @SharyRajpoot ,

If you can file a new issue following our New Issue Guidelines with additional details around your issue, we can try to help you with your issue. 

Moreover, could you please share the code snippet and overall GDPR logic you apply on your app?

Thanks,
MoPub Team

**Comment by SharyRajpoot** [comment]
This is my logic which determines if GDRP is applicable to the user or not.

if ( !MoPub.isSdkInitialized( ) ) {
     val sdkConfiguration = SdkConfiguration.Builder("KEY")
                //.withLogLevel(MoPubLog.LogLevel.DEBUG)
                .withMediatedNetworkConfiguration(AdColonyAdapterConfiguration::class.java.name, adColonyConfigs)
                .withMediatedNetworkConfiguration(GooglePlayServicesAdapterConfiguration::class.java.name, configMap)
                .build()
                
                MoPub.initializeSdk(context, sdkConfiguration, object : SdkInitializationListener {
                override fun onInitializationFinished() {
                       val personalInfoManager = PersonalInfoManager( context, "KEY", null )
                       if ( personalInfoManager.gdprApplies() == true || personalInfoManager.shouldShowConsentDialog( ) ) {
                            val consentDialog = ConsentDialog(personalInfoManager)
                            consentDialog.show((context as MainActivity).supportFragmentManager, "CONSENT_DIALOG")
                        }
                }
}

But these two functions gdprApplies( ) & shouldShowConsentDialog( )  
always return false wheater I am in GDRP region...means through VPN.
even I have clear all cache and re-open the app.

**Comment by SharyRajpoot** [comment]
Ok buddy I got the problem that is causing the consent dialog not appearing...
the thing is when I start my app the first time the app asked for some permission related to the app and meanwhile MoPub SDK detects that GDRP is applicable to that user...
So my current activity contains a fragment that is currently shown in it...and then immediately the app asked for run time permission which means on activity lifecycle onSavedInstanceState( ) was called and it just saved the activity state...
and meanwhile, MoPub tries to display a custom dialog that I have to create for asking consent permission, so when it tries to load/show it, it's failed because the activity state is already saved.
in short, I got this error...

**Can not perform this action after onSaveInstanceState**

which is causing the MoPub not showing the dialog but I noticed this when I completely stop the app and re-run it then the MoPub consent dialog is successfully shown because at that time app is not asking for any run time permissions.

The above code is working fine and correct. 👍 :-)

**Comment by hmertkahraman** [comment]
Thank you very much for the update.
